plugin/src/App/init.lua

```diff
@@ -462,6 +465,31 @@ function App:startSession()
 			return "Accept"
 		end
 
+		local confirmationBehavior = Settings:get("confirmationBehavior")
+		if confirmationBehavior == "Initial" then
+			-- Only confirm if we haven't synced this project yet this session
+			if self.knownProjects[serverInfo.projectName] then
+				Log.trace("Accepting patch without confirmation because project has already been connected and behavior is set to Initial")
+				return "Accept"
+			end
+		elseif confirmationBehavior == "Large Changes" then
+			-- Only confirm if the patch impacts many instances
+			local instanceThreshold = 5
```

Comment:
This feels like it should be a constant moved up scope!

Reply:
Maybe even as a setting?

Reply:
I originally had it inline and decided to make it a variable so it would have a clear name for future maintainers. I'm open to making it a user configurable setting though!

Reply:
I feel like a setting is the way to go, yeah!